### PR TITLE
Add support to run build on Container image used by release pipeline

### PR DESCRIPTION
**Pull request by monusingh-1**
### Description
CI to run on top of the Build Docker Images from production distribution pipeline.
 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/1228
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saikaranam-amazon — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,6 +11,66 @@ on:
...
+        run: |
+          crane version
+          CI_IMAGE_VERSION=`opensearch-build/docker/ci/get-ci-images.sh -p centos7 -u opensearch -t build | head -1`
+          echo $CI_IMAGE_VERSION
+          echo "ci-image-version-linux=$CI_IMAGE_VERSION" >> $GITHUB_OUTPUT
+
+  Build-CCR-Linux:
+    strategy:
+      matrix:
+        java: [11, 17]
```
Comment (ankitkala) [inline .github/workflows/build.yml:40]: Are we building for both java 11 & 17 now? earlier we were doing only 17.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -11,6 +11,66 @@ on:
...
+
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Run build
+        # switching the user, as OpenSearch cluster can only be started as root/Administrator on linux-deb/linux-rpm/windows-zip.
+        run: |
+          chown -R opensearch.opensearch `pwd`
```
Comment (ankitkala) [inline .github/workflows/build.yml:66]: We have 2 separate workflow `build` & `build-and-test`. Do we need to change both? 
We're also running tests in this one which we weren't doing earlier.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -11,6 +11,66 @@ on:
...
+          su opensearch -c "whoami && java -version && ./gradlew build"
+
+      - name: Upload Coverage Report
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+
   build:
```
Comment (ankitkala) [inline .github/workflows/build.yml:74]: Do we still need the steps below?

File: .github/workflows/build.yml (hunk 1 above)
Comment (monusingh-1) [reply .github/workflows/build.yml:40]: We are running tests on both 11 and 17 so we can build on both

File: .github/workflows/build.yml (hunk 2 above)
Comment (monusingh-1) [reply .github/workflows/build.yml:66]: On the issue, currently they require (specifically plugin build). to run on container image.

File: .github/workflows/build.yml (hunk 3 above)
Comment (monusingh-1) [reply .github/workflows/build.yml:74]: GitHub Actions only support LINUX docker container at the time. Support for Windows and macOS container will be added later on.

**Comment by peterzhuamazon** [comment]
Hi @ankitkala @monusingh-1 I am adding some changes directly to this PR.
Since we have a even better approach right now.

Will commit soon.

Thanks

**Comment by peterzhuamazon** [comment]
Ready for review.

File: .github/workflows/build.yml (hunk 2 above)
Comment (monusingh-1) [reply .github/workflows/build.yml]: Change is done to user containers for both build and test

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -11,6 +11,45 @@ on:
...
+      product: opensearch
+
+  build-linux:
+    strategy:
+      matrix:
+        java: [11, 17]
+
+    name: Build and Test CCR Plugin on Linux
+    runs-on: ubuntu-latest
+    needs: Get-CI-Image-Tag
```
Comment (ankitkala) [inline .github/workflows/build.yml:26]: Not sure if we're adding any value by this extra workflow `Build on ubuntu with ci image`. We're already testing the same under `build-and-test`. This one can be removed.

**Review by ankitkala — APPROVED** [review]

**Review by saikaranam-amazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Hi @ankitkala @monusingh-1 thanks for merging.
Can we manually backport this to 2.x?

Thanks!

**Pull request merged**